packages/promises/promise.pony

```diff
@@ -69,6 +69,168 @@ actor Promise[A: Any #share]
     _attach(consume attach)
     promise
 
+  fun tag flatten_next[B: Any #share](
+    fulfill: Fulfill[A, Promise[B]],
+    rejected: Reject[Promise[B]] = RejectAlways[Promise[B]])
+    : Promise[B]
+  =>
+    """
+    Chain a promise after this one and unwrap the promise returned from this
+    one.
+
+    `flatten_next` is a companion to `next`. It operates in an identical
+    fashion except for the type of the fulfilled promise. Whereas `next` takes
+    a function that returns a type `B`, `flatten_next` takes a function that
+    returns `Promise[B]`.
+
+    Why is `flatten_next` valuable given that next could take a `B` that is of
+    a type like `Promise[String]`? Let's start with some code to demonstrate the
+    problem that arises when returning `Promise[Promise[B]]` from `next`.
+
+    Let's say we have a library for accessing the GitHub REST API:
+
+    ```pony
+    class GitHub
+      new create(personal_access_token: String)
+
+      fun get_repo(repo: String): Promise[Repository]
```

Comment:
Same here re: `Repository` -> `Repo`

Reply:
It was intentional to make it fit on one line

Reply:
Ok, I'd suggest renaming the type in the signature to `Repo` since that's what we use in the rest of the example. Right now the type here and the types in the rest of the example don't match.

Reply:
updated